sympy/polys/rings.py

```diff
@@ -2531,15 +2531,140 @@ def compose(f, x, a=None):
 
         return poly
 
+
+    def coeff_wrt(self, x, dg):
+        """
+        Coefficient of ``f`` with respect to ``x**deg``.
+
+        Treating ``f`` as a univariate polynomial in ``x`` this finds the
+        coefficient of ``x**deg`` as a polynomial in the other generators.
+
+        Parameters
+        ==========
+
+        p : sympy.Poly
+            The polynomial to compute the expression for.
+        x : generator or generator index
+            The generator or generator index to compute the expression for.
+        dg : int
+            The degree of the monomial to compute the expression for.
+
+        Returns
+        =======
+
+        sympy.Expr
+            The expression of the polynomial ``f`` at the given symbol and degree.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> R, x, y, z = ring("x, y, z", ZZ)
+        >>> p = 2*x**4 + 3*y**4 + 10*z**2 + 10*x*z**2
+        >>> dg = 2
+        >>> p.coeff_wrt(2, dg) # Using the generator index
+        10*x + 10
+        >>> p.coeff_wrt(z, 2) # Using the generator
+        10*x + 10
```

Comment:
Remove the `dg = 2`.

Reply:
Or otherwise use it in both examples.

Reply:
It would be good to include an example using `coeff` here to show the difference e.g.:
```python
>>> p.coeff(z**2)
10
```
Also it would be good to have a see also section referring to `coeff` and possibly other related methods.